homeassistant/components/dominos.py

```diff
@@ -0,0 +1,251 @@
+"""
+Support for Dominos Pizza ordering.
+
+The Dominos Pizza component ceates a service which can be invoked to order
+from their menu
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/dominos/.
+"""
+import asyncio
+import logging
+import time
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components import http
+from homeassistant.core import callback
+from homeassistant.helpers.entity import Entity, async_generate_entity_id
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+# The domain of your component. Should be equal to the name of your component.
+DOMAIN = 'dominos'
+ENTITY_ID_FORMAT = DOMAIN + '.{}'
+
+ATTR_COUNTRY = 'country_code'
+ATTR_FIRST_NAME = 'first_name'
+ATTR_LAST_NAME = 'last_name'
+ATTR_EMAIL = 'email'
+ATTR_PHONE = 'phone'
+ATTR_ADDRESS = 'address'
+ATTR_ORDERS = 'orders'
+ATTR_SHOW_MENU = 'show_menu'
+ATTR_ORDER_ENTITY = 'order_entity_id'
+ATTR_ORDER_NAME = 'name'
+ATTR_ORDER_CODES = 'codes'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=10)
+MIN_TIME_BETWEEN_STORE_UPDATES = 1800
+
+REQUIREMENTS = ['pizzapi==0.0.2']
+
+DEPENDENCIES = ['http']
+
+_ORDERS_SCHEMA = vol.Schema({
+    vol.Required(ATTR_ORDER_NAME): cv.string,
+    vol.Required(ATTR_ORDER_CODES): vol.All(cv.ensure_list, [cv.string]),
+})
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(ATTR_COUNTRY): cv.string,
+        vol.Required(ATTR_FIRST_NAME): cv.string,
+        vol.Required(ATTR_LAST_NAME): cv.string,
+        vol.Required(ATTR_EMAIL): cv.string,
+        vol.Required(ATTR_PHONE): cv.string,
+        vol.Required(ATTR_ADDRESS): cv.string,
+        vol.Optional(ATTR_SHOW_MENU): cv.boolean,
+        vol.Optional(ATTR_ORDERS): vol.All(cv.ensure_list, [_ORDERS_SCHEMA]),
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+# pylint: disable=broad-except
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Set up is called when Home Assistant is loading our component."""
+    dominos = Dominos(hass, config)
+
+    component = EntityComponent(_LOGGER, DOMAIN, hass)
+    hass.data[DOMAIN] = {}
+    entities = []
+
+    hass.services.async_register(DOMAIN, 'order', dominos.handle_order)
+
+    if config[DOMAIN].get(ATTR_SHOW_MENU):
+        # pylint: disable=not-an-iterable
+        yield from dominos.show_menu(hass)
+        hass.http.register_view(DominosProductListView)
+        yield from hass.components.frontend.async_register_built_in_panel(
```

Comment:
I've closed the PR for the built-in panel. This is too much.

Reply:
If you want to build it out to a food component with platforms that offer a menu and allow ordering, a panel would be ok.